Source/WebCore/css/parser/CSSParserImpl.cpp

```diff
@@ -593,6 +603,157 @@ RefPtr<StyleRuleFontFace> CSSParserImpl::consumeFontFaceRule(CSSParserTokenRange
     return StyleRuleFontFace::create(createStyleProperties(m_parsedProperties, m_context.mode));
 }
 
+static std::pair<FontFeatureValuesType, unsigned> fontFeatureValuesTypeMappings(CSSAtRuleID id)
+{
+    switch (id) {
+    // Note that values higher than 99 are valid for styleset, but don't map to any OpenType values and are ignored.
+    // cf https://developer.mozilla.org/en-US/docs/Web/CSS/@font-feature-values
+    case CSSAtRuleStyleset: return { FontFeatureValuesType::Styleset, 99 }; 
+    case CSSAtRuleStylistic: return { FontFeatureValuesType::Stylistic, 1 };
+    case CSSAtRuleCharacterVariant: return { FontFeatureValuesType::CharacterVariant, 2 };
+    case CSSAtRuleSwash: return { FontFeatureValuesType::Swash, 1 };
+    case CSSAtRuleOrnaments: return { FontFeatureValuesType::Ornaments, 1 };
+    case CSSAtRuleAnnotation: return { FontFeatureValuesType::Annotation, 1 };
+    default:
+        ASSERT_NOT_REACHED();
+        return { };
+    }
+}
+
+RefPtr<StyleRuleFontFeatureValuesBlock> CSSParserImpl::consumeFontFeatureValuesRuleBlock(CSSAtRuleID id, CSSParserTokenRange prelude, CSSParserTokenRange range)
+{
+    // <feature-value-block> = <font-feature-value-type> { <declaration-list> }
+    // <font-feature-value-type> = @stylistic | @historical-forms | @styleset | @character-variant | @swash | @ornaments | @annotation
+
+    // Prelude should be empty.
+    if (!prelude.atEnd())
+        return { };
+    
+    // Block should be present.
+    if (range.atEnd())
+        return { };
+    
+    if (m_observerWrapper) {
+        m_observerWrapper->observer().startRuleHeader(StyleRuleType::FontFeatureValuesBlock, m_observerWrapper->startOffset(prelude));
+        m_observerWrapper->observer().endRuleHeader(m_observerWrapper->endOffset(prelude));
+        m_observerWrapper->observer().startRuleBody(m_observerWrapper->previousTokenStartOffset(range));
+        m_observerWrapper->observer().endRuleBody(m_observerWrapper->endOffset(range));
+    }
+
+    auto [type, maxValues] = fontFeatureValuesTypeMappings(id);
+
+    auto consumeTag = [](CSSParserTokenRange range, uint maxValues) -> std::optional<Tag> {
```

Comment:
AppleWin and WinCairo EWS reported a compilation error.
> C:\cygwin\home\buildbot\worker\Windows-EWS\build\Source\WebCore\css/parser/CSSParserImpl.cpp(645,53): error C2760: syntax error: 'identifier' was unexpected here; expected 'type specifier' (compiling source file C:\cygwin\home\buildbot\worker\Windows-EWS\build\WebKitBuild\Release\DerivedSources\WebCore\unified-sources\UnifiedSource-2f84417a-9.cpp) [C:\cygwin\home\buildbot\worker\Windows-EWS\build\WebKitBuild\Release\Source\WebCore\WebCore.vcxproj]
Replacing `uint` with `unsigned` solves the problem.

Reply:
Thank you